lib/oban/plugins/cron.ex

```diff
@@ -167,16 +169,31 @@ defmodule Oban.Plugins.Cron do
 
   defp validate_opt!(_opt), do: :ok
 
-  defp valid_crontab?({expression, worker, opts}) do
+  defp validate_crontab!({expression, worker, opts}) do
     %Expression{} = Expression.parse!(expression)
 
-    Code.ensure_loaded?(worker) and
-      function_exported?(worker, :perform, 1) and
-      Keyword.keyword?(opts)
+    unless Code.ensure_loaded?(worker) do
+      raise ArgumentError, "#{inspect(worker)} not found or can't be loaded"
+    end
+
+    unless function_exported?(worker, :perform, 1) do
+      raise ArgumentError, "#{inspect(worker)} does not implement `perform/1` callback"
+    end
+
+    unless Keyword.keyword?(opts) do
+      raise ArgumentError, "options for #{inspect(worker)} must be as a keyword list"
+    end
+  end
+
+  defp validate_crontab!({expression, worker}) do
+    validate_crontab!({expression, worker, []})
   end
 
-  defp valid_crontab?({expression, worker}), do: valid_crontab?({expression, worker, []})
-  defp valid_crontab?(_crontab), do: false
+  defp validate_crontab!(invalid) do
+    raise ArgumentError,
+          "#{inspect(invalid)} is invalid cronjob declaration, expected {expression, worker} or " <>
+            "{expression, worker, options} tuples"
```

Comment:
Putting the format first and the "got" part last matches the format of the other errors better:

```suggestion
    raise ArgumentError,
          "expected crontab entry to be an {expression, worker} or " <>
            "{expression, worker, options} tuple, got: #{inspect(invalid)}"
```